### ISSUE TITLE
An incorrect `set errexit` command suppresses warnings

### ISSUE THREAD

**Issue by jamesderlin** [koalaman/shellcheck#2233]
shellcheck seems to assume that any line of the form `set ... errexit` enables automatic exit on failure, so mistakenly using `set errexit` in a script instead of `set -o errexit` will go uncaught and will undesirably suppress other warnings.

#### For bugs
- SC2164
- My shellcheck version (`shellcheck --version` or "online"): 0.7.0 and online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
set errexit
cd foo
```

#### Here's what shellcheck currently says:
Nothing.

#### Here's what I wanted or expected to see:
```
cd foo
^----^ SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

Did you mean: 
cd foo || exit
```
